ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -708,40 +708,49 @@ let query_domain_workgroup ~domain =
     workgroup_from_server kdc |> Result.get_ok
   with _ -> raise (Auth_service_error (E_LOOKUP, err_msg))
 
-let config_winbind_damon ~domain ~workgroup ~netbios_name =
+let config_winbind_damon ?(workgroup = "") ?(netbios_name = "") domain =
   let open Xapi_stdext_unix in
   let smb_config = "/etc/samba/smb.conf" in
   let conf_contents =
-    String.concat "\n"
-      [
-        "# autogenerated by xapi"
-      ; "[global]"
-      ; "kerberos method = secrets and keytab"
-      ; Printf.sprintf "realm = %s" domain
-      ; "security = ADS"
-      ; "template shell = /bin/bash"
-      ; "winbind refresh tickets = Yes"
-      ; "winbind enum groups = no"
-      ; "winbind enum users = no"
-      ; "winbind scan trusted domains = yes"
-      ; "winbind use krb5 enterprise principals = yes"
-      ; Printf.sprintf "winbind cache time = %d" !Xapi_globs.winbind_cache_time
-      ; Printf.sprintf "machine password timeout = %d"
-          !Xapi_globs.winbind_machine_pwd_timeout
-      ; Printf.sprintf "kerberos encryption types = %s"
-          (Kerberos_encryption_types.Winbind.to_string
-             !Xapi_globs.winbind_kerberos_encryption_type
-          )
-      ; Printf.sprintf "workgroup = %s" workgroup
-      ; Printf.sprintf "netbios name = %s" netbios_name
-      ; "idmap config * : range = 3000000-3999999"
-      ; Printf.sprintf "idmap config %s: backend = rid" domain
-      ; Printf.sprintf "idmap config %s: range = 2000000-2999999" domain
-      ; Printf.sprintf "log level = %s" (debug_level ())
-      ; "idmap config * : backend = tdb"
-      ; "" (* Empty line at the end *)
-      ]
+    match domain with
+    | Some dom ->
+        String.concat "\n"
+          [
+            "# autogenerated by xapi"
+          ; "[global]"
+          ; "kerberos method = secrets and keytab"
+          ; Printf.sprintf "realm = %s" dom
+          ; "security = ADS"
+          ; "template shell = /bin/bash"
+          ; "winbind refresh tickets = yes"
+          ; "winbind enum groups = no"
+          ; "winbind enum users = no"
+          ; "winbind scan trusted domains = yes"
+          ; "winbind use krb5 enterprise principals = yes"
+          ; Printf.sprintf "winbind cache time = %d"
+              !Xapi_globs.winbind_cache_time
+          ; Printf.sprintf "machine password timeout = %d"
+              !Xapi_globs.winbind_machine_pwd_timeout
+          ; Printf.sprintf "kerberos encryption types = %s"
+              (Kerberos_encryption_types.Winbind.to_string
+                 !Xapi_globs.winbind_kerberos_encryption_type
+              )
+          ; Printf.sprintf "workgroup = %s" workgroup
+          ; Printf.sprintf "netbios name = %s" netbios_name
```

Comment:
Is it OK to write `workgroup = ` in the file? (same for netbios)

Reply:
why do you think it is a problem?

Reply:
The main reason is that having empty values may be handled in a way different from not having it defined at all.

I see in the manpage that some default are defined with an empty value, so it seems this should be OK.
> Default: `add group script = `